pricing/src/main/scala/dev/atedeg/mdm/pricing/Types.scala

```diff
@@ -0,0 +1,80 @@
+package dev.atedeg.mdm.pricing
+
+import java.time.LocalDateTime
+import java.util.UUID
+
+import cats.data.NonEmptyList
+
+import dev.atedeg.mdm.products.Product
+import dev.atedeg.mdm.utils.*
+
+/**
+ * A [[Product product]] with its ordered [[Quantity quantity]].
+ */
+final case class IncomingOrderLine(quantity: Quantity, product: Product)
```

Comment:
There is no ambiguity here, so the "ordered" part can be seen as superfluous. Also, if this change gets made, we should also change the Quantity in the client orders BC, since the same reasoning could be applied there.